pkg/solana/fees/block_history_test.go

```diff
@@ -15,82 +16,442 @@ import (
+		cfg.On("ComputeUnitPriceMax").Return(max)
+		assert.Equal(t, uint64(100), estimator.BaseComputeUnitPrice(), "Price should not change when median calculation fails")
+	})
+
+	t.Run("Failed to Get Client", func(t *testing.T) {
+		// Setup
+		rwFailLoader := utils.NewLazyLoad(func() (client.ReaderWriter, error) {
+			// Return error to simulate failure to get client
+			return nil, fmt.Errorf("fail client load")
+		})
+		cfg := cfgmock.NewConfig(t)
+		setupConfigMock(cfg, defaultPrice, min, max, pollPeriod, depth)
+		lgr, _ := logger.TestObserved(t, zapcore.DebugLevel)
+		ctx := tests.Context(t)
+		estimator := initializeEstimator(t, rwFailLoader, cfg, lgr, ctx)
+
+		// Call calculatePrice and expect an error
+		// Ensure the price remains unchanged
+		require.Error(t, estimator.calculatePrice(ctx), "Expected error when getting client fails")
+		cfg.On("ComputeUnitPriceMax").Return(max)
+		assert.Equal(t, defaultPrice, estimator.BaseComputeUnitPrice(), "Price should remain at default when client fails")
+	})
+}
+
+func TestBlockHistoryEstimator_MultipleBlocks(t *testing.T) {
+	// helpers vars for tests
+	min := uint64(100)
+	max := uint64(100_000)
+	depth := uint64(3)
+	defaultPrice := uint64(100)
+	pollPeriod := 3 * time.Second
+
+	// Read multiple blocks from JSON file
+	testBlocks := readMultipleBlocksFromFile(t, "./multiple_blocks_data.json")
+	require.GreaterOrEqual(t, len(testBlocks), int(depth), "Not enough blocks in JSON to match BlockHistorySize")
+
+	// Extract slots and compute unit prices from the blocks
+	// We'll consider the last 'BlockHistorySize' blocks
+	var testSlots []uint64
+	var testPrices []ComputeUnitPrice
+	startIndex := len(testBlocks) - int(depth)
+	testBlocks = testBlocks[startIndex:]
+	for _, block := range testBlocks {
+		// extract compute unit prices and get median from each block
+		slot := block.ParentSlot + 1
+		testSlots = append(testSlots, slot)
+		feeData, err := ParseBlock(block)
+		require.NoError(t, err, "Failed to parse block at slot %d", slot)
+		require.NotEmpty(t, feeData.Prices, "No compute unit prices found in block at slot %d", slot)
+		medianPrice, err := mathutil.Median(feeData.Prices...)
+		testPrices = append(testPrices, medianPrice)
+	}
+	testSlotsResult := rpc.BlocksResult(testSlots)
+	multipleBlocksMedian, _ := mathutil.Median(testPrices...)
 
-	// happy path
+	t.Run("Successful Estimation", func(t *testing.T) {
+		// Setup
+		rw := clientmock.NewReaderWriter(t)
+		rwLoader := utils.NewLazyLoad(func() (client.ReaderWriter, error) {
+			return rw, nil
+		})
+		cfg := cfgmock.NewConfig(t)
+		ctx := tests.Context(t)
+		setupConfigMock(cfg, defaultPrice, min, max, pollPeriod, depth)
+		rw.On("SlotHeight", mock.Anything).Return(testSlots[len(testSlots)-1], nil)
+		rw.On("GetBlocksWithLimit", mock.Anything, mock.Anything, mock.Anything).
+			Return(&testSlotsResult, nil)
+		for i, slot := range testSlots {
+			rw.On("GetBlock", mock.Anything, slot).
+				Return(testBlocks[i], nil)
+		}
```

Comment:
Same as the latest block BHE tests, could we move this ReaderWriter mock to the parent test? And just create new mocks for the tests for RPC failures?